contracts/proxies/GnosisSafeProxyFactory.sol

```diff
@@ -101,7 +106,45 @@ contract GnosisSafeProxyFactory {
         bytes calldata initializer,
         uint256 saltNonce
     ) external returns (GnosisSafeProxy proxy) {
-        proxy = deployProxyWithNonce(_singleton, initializer, saltNonce);
+        bytes32 salt = keccak256(abi.encodePacked(keccak256(initializer), saltNonce));
+        proxy = deployProxy(_singleton, salt);
+        revert(string(abi.encodePacked(proxy)));
+    }
+
+    /// @dev Allows to get the address for a new proxy contact created via `createChainSpecificProxyWithNonce`
+    ///      This method is only meant for address calculation purpose when you use an initializer that would revert,
+    ///      therefore the response is returned with a revert. When calling this method set `from` to the address of the proxy factory.
+    /// @param _singleton Address of singleton contract.
+    /// @param initializer Payload for message call sent to new proxy contract.
+    /// @param saltNonce Nonce that will be used to generate the salt to calculate the address of the new proxy contract.
+    function calculateCreateChainSpecificProxyWithNonceAddress(
```

Comment:
ahh I know why you added this and it is not a blocker, but I would not add any more methods that use the `revert` approach to return data. It doesn't work well with all nodes (as they all return revert messages differently. If we want to we could refactor this into a custom error with a bytes param ... but that is not available at the solidity version we use afaik.

That being said I would remove both method to calculate the address completely (they provide no benefit). Also maybe we should remove the `proxyRuntimeCode` as it provides little value (as on-chain information) and breaks zkSync support.

Reply:
I agree with reverting methods, they can really behave in an unexpected manner, see https://github.com/NomicFoundation/hardhat/issues/3438

regarding `proxyRuntimeCode` i'd like to hear Uxio's opinion (iirc methods were added because of him) but he's on vacation until Jan 16th

Reply:
This was used in the relay service which we deprecated, therefore I would optimistically remove the methods, worst case we add them back later :P

Reply:
Done, please check

Reply:
I also meant https://github.com/safe-global/safe-contracts/pull/468/files#diff-7a756ca403cd3005982b1a2b902059a02ed81661eae64998b42c8d51972ea5f0L28-L30 with my comment ;)